realm/realm-annotations-processor/src/test/java/io/realm/processor/RealmProcessorTest.java

```diff
@@ -278,7 +278,7 @@ public void compileCustomAccessor() {
     @Test
     public void compileIndexTypes() throws IOException {
         final String[] validIndexFieldTypes = {"byte", "short", "int", "long", "boolean", "String", "java.util.Date",
-                "Byte", "Short", "Integer", "Long", "Boolean"};
+                "Byte", "Short", "Integer", "Long", "Boolean", "org.bson.types.ObjectId", "java.util.UUID"};
```

Comment:
Probably not for this PR, but bookkeeping/correctness of all these test would be improved significantly if we introduced `RealmFieldType.indexable()`, etc. and had the test iterate the `RealmFieldType.values()`. If not needed in the public API we could maybe add some utilities for test with the same concept. But seems like there are plenty of places where it could be convenient internally too ex. https://github.com/realm/realm-java/pull/7152/files#diff-6d4cb257e0e9ac7445f8cce3a43be3a9c690525ec6268d5ddb96fb80d157ce4bR58